### ISSUE TITLE
"Don't load project until interacted with" addon

### ISSUE THREAD

**Issue by towerofnix** [ScratchAddons/ScratchAddons#3566]
### Is your request related to a problem? Please describe.

On devices with lower RAM availability, or just when viewing large projects, serious slowdown can be caused by downloading project assets and loading them into memory, even if they aren't run. This can be pretty annoying when viewing or replying to comments on such projects you've made yourself!

### Describe the solution you'd like

A pretty simple addon whose behavior I don't think is included in other addons: don't start downloading projects until you've clicked a "start loading" button laid over the stage.

### Describe alternatives you've considered

Replying via Scratch Messaging is a current way to avoid loading the project page to view comments, but not everyone uses that addon, and a general feature to keep projects from automatically using memory would have other uses too—making quick tweaks to notes and credits for example.

**Comment by Samq64** [comment]
This is only a serious problem with mobile devices which don't support browser extensions. (unless you don't mind using Kiwi) But still, this could be useful.

**Comment by towerofnix** [comment]
@Samq64 Yeah, though on my system it honestly doesn't take *that* large of a project to cause noticeable stuttering. A project which takes 3GB to load when I have 300MB non-allocated means distinct lag as the computer takes things out of its cache (especially if I try to do anything else while that project tab is still open—those 3GB *aren't* being used for anything normally cached anymore, and even once the tab's closed it has to be recompiled/loaded/etc).

In general my computer doesn't reserve memory for core processes very well, so larger projects just freeze the whole screen from updating at all. It's not too common but it's annoying enough for an addon to help. And devices with 4GB or even less RAM probably have it significantly worse off than myself!

**Comment by lisa-wolfgang** [comment]
This would also be useful to help reduce data usage for people using limited data plans.

**Comment by DNin01** [comment]
We could also just only use this if the user is going to a comment on a project (by clicking the link from Messages). Such a feature could get the name "Do not automatically load project when opening comment" or something like that.

**Comment by towerofnix** [comment]
@DNin01 That's a good point, but I think it has more uses than only when viewing comments. Sometimes I want to view or revise the instructions or notes & credits of a project and I can't really do that without loading the whole project—which can be especially annoying since the site tends to lag out when loading a ton of large assets at once, and then I can't make edits at all (or necessarily even scroll those areas, load comments/replies, etc).

I think a checkbox which you could specifically toggle, like "Only wait when opening comment", could work for those who are only concerned with replying to comments and still generally like projects to auto-load.

**Comment by WorldLanguages** [comment]
> UI isn't really a problem. The technical side of this is more challenging.

How would we stop the project from loading, and resume it at a later point? With JavaScript, I suppose. Anyone have any ideas on the specific way we could achieve this?

**Comment by CST1229** [comment]
> How would we stop the project from loading, and resume it at a later point? With JavaScript, I suppose. Anyone have any ideas on the specific way we could achieve this?

Infinitely delay the project.json request (and if necessary, assets too) using fetch().
